src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java

```diff
@@ -49,15 +49,17 @@ public void start(Method method, Object[] data) {
       NewDriverProvider.setUnstablePageLoadStrategy(true);
     }
 
-    String onDriver = method.getAnnotation(Execute.class).allowedDriver();
-    if (onDriver.length() > 0 & !onDriver.equalsIgnoreCase(Configuration.getBrowser())) {
-      throw new SkipException(
-          "The test can not be run on driver " + Configuration
-              .getBrowser() + ". The test is restricted to driver " + onDriver);
-    } else {
-      startBrowser();
+    if (method.isAnnotationPresent(Execute.class)) {
+      String onDriver = method.getAnnotation(Execute.class).allowedDriver();
+      if (onDriver.length() > 0 & !onDriver.equalsIgnoreCase(Configuration.getBrowser())) {
+        throw new SkipException(
+            "The test can not be run on driver " + Configuration
+                .getBrowser() + ". The test is restricted to driver " + onDriver);
+      }
     }
 
+    startBrowser();
+
     if (method.isAnnotationPresent(Execute.class)) {
```

Comment:
consider combine those two ifs:
https://github.com/Wikia/selenium-tests/pull/847/files#diff-d7d0f5104feb18b29563c05b2d1f0e7cR52
and
https://github.com/Wikia/selenium-tests/pull/847/files#diff-d7d0f5104feb18b29563c05b2d1f0e7cR63